### PR TITLE
Feat: in world camera movement

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?

Implements transition to the in-world camera from different cinemachine states and control of its movement:

- Move Camera: Click and Drag
- Zoom: mouse wheel
- UP / DOWN: R / F
- FORWARD / BACKWARD / STRAFE LEFT / STRAFE RIGHT: W, S, A, D
- Rotate: MOUSE
- Nametags Switch: N
- Open/Close Camera: C
- Take a Picture: space
- LeftShift (hold): translation speed-up

I ported movement functionality from the old client. It is implemented via Cinemachine, but in this case uses actual player camera instead of creating new one. 
Includes bug fix for 90/-90 degree rotation that is presented in the old client. 

## How to test the changes?

1. Launch the explorer with the `--camera-reel` CLI argument
2. Press C to enter camera mode - check different edge cases for this  transitions (exploratory testing)
3. Use WASD, RF and mouse to move around
4. Verify you cannot pass through colliders (should be same as for your avatar)
5. Verify you camera movement is limited by around 16 metters around the player and never goes below zero ground level
6. Check the zoom via mouse scroll - play around with transitions to different states with zoom changed (exploratory testing)
7. Optionally: take a picture via Spacebar. You can check that picture by going to the old client (it should appear after a while in the CameraReel Gallery there)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: Explorer/Assets/DCL/Character/CharacterCamera/Systems/UpdateCinemachineBrainSystem.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        public UpdateCinemachineBrainSystem(World world) : base(world) { }
+
+        protected override void Update(float t)
+        {
+            ManualBrainUpdateQueryQuery(World);
+        }
+
+        [Query]
+        [None(typeof(CameraLookAtIntent))]
```
[the remark this answers is not in the document]
Comment (popuz) [reply Explorer/Assets/DCL/Character/CharacterCamera/Systems/UpdateCinemachineBrainSystem.cs:23]: nope. That was exectly the reason of extracting it to another system, so I don't need to call  `cinemachinePreset.Brain.ManualUpdate();` separately for `InWorldCamera`. If I would have more time, I would prefer to split even more all the logic of the Cinemachine to smaller ECS systems that will query properly and won't need additional checks. In that case probably other parts would become also more re-usable. But for now, I just extracted only the brain update

**Comment by popuz** [comment]
Instead of classical ECS query approach, I went here with the `TryGet`. While I would prefer to use ECS approach, main concern was the performance, since empty queries and queries for single entity still performs worse than `TryGet`.  I made performance measurements in the build with the `Stopwatch` for more precise results. First measurement was comparing  Query
```
        [Query]
        private void EmptyInputCheck(in InWorldCameraInput input)
        {
            takeScreenshot = input.TakeScreenshot;
        }
```
 versus 
```
World.TryGet<InWorldCameraInput>(camera, out var newInput);
takeScreenshot = newInput.TakeScreenshot;
```
Results that `TryGet `is at least twice faster for empty call (no component presented) and x2.5 faster when component presented: 
`0.0025/0.0037 ms` vs `0.005/0.0068 ms` and  `0.004 ms` vs  `0.01 ms` respectively (approximate values).

 Then I compared it with the inlineQuery method, hoping to get more performance from it
 ```
      World.InlineQuery<EmitInput, InWorldCameraInput>(in query, ref velUpdate);
 
        public struct EmitInput : IForEach<InWorldCameraInput>
        {
            public bool takeScreenshot;

            [MethodImpl(MethodImplOptions.AggressiveInlining)]
            public void Update(ref InWorldCameraInput input) {
                takeScreenshot = input.TakeScreenshot;
            }
        }
```

Surprisingly,  `inlineQuery `performed slightly worse than normal `Query`. When no component presented it was `0.0085 ms` (vs  `0.005/0.0068 ms`  for Query and `0.0025/0.0037 ms` for TryGet)

While it is a small numbers for one such system, creating 10 systems already can pill it up to approximately `0.10 ms` .

**Review by dalkia — APPROVED** [review]
